openwillis/measures/text/speech_attribute.py

```diff
@@ -114,48 +113,29 @@ def filter_transcribe(json_conf, measures, speaker_label=None):
     ------------------------------------------------------------------------------------------------------
     """
     item_data = json_conf["results"]["items"]
-
-    # make a dictionary to map old indices to new indices
-    item_data = cutil.create_index_column(item_data, measures)
+    
+    for i, item in enumerate(item_data): # create_index_column
+        item[measures["old_index"]] = i
 
     # extract text
-    text = " ".join(
-        [
-            item["alternatives"][0]["content"]
-            for item in item_data
-            if "alternatives" in item
-        ]
-    )
-
-    # phrase-split
+    text = " ".join([item["alternatives"][0]["content"] for item in item_data if "alternatives" in item])
     phrases, phrases_idxs = cutil.phrase_split(text)
 
-    # turn-split
-    turns = []
-    turns_idxs = []
-
     if speaker_label is not None:
+        turns_idxs, turns = cutil.filter_speaker_aws(item_data, speaker_label)
+    else:
+        turns_idxs, turns = [], []
 
-        turns_idxs, turns, phrases_idxs, phrases = cutil.filter_speaker(
-            item_data, speaker_label, turns_idxs, turns, phrases_idxs, phrases
-        )
-
-    # entire transcript - by joining all the phrases
     text = " ".join(phrases)
-
-    # filter json to only include items with start_time and end_time
-    filter_json = cutil.filter_json_transcribe(item_data, speaker_label, measures)
-
-    # extract words
+    filter_json = cutil.filter_json_transcribe_aws(item_data, speaker_label, measures)
     words = [word["alternatives"][0]["content"] for word in filter_json]
 
-    text_list = [words, phrases, turns, text]
+    text_list = [words, turns, text]
     text_indices = [phrases_idxs, turns_idxs]
```

Comment:
```suggestion
    text_indices = [turns_idxs]
```

or not even list, since we are removing phrases

Reply:
phrases_idxs were used in the underline logic to compute word level pause features. (function: get_pause_feature_word)

I guess you added this function in the past, i was not sure how to handle it, so kept  phrases_idxs to compute word level pause features

Reply:
Ok yeah I remember, so we use them so that we set to NaN the pre word pauses that are in the start of phrases (since they are pre phrase pauses), but if we are not outputting phrase level summary statistics, should we remove that and have them be pre-word pauses?